src/main/java/org/spongepowered/common/data/utils/items/SpongeAuthorProcessor.java

```diff
@@ -0,0 +1,131 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.utils.items;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spongepowered.common.data.DataTransactionBuilder.fail;
+import static org.spongepowered.common.data.DataTransactionBuilder.successNoData;
+
+import com.google.common.base.Optional;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagString;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataPriority;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.manipulator.item.AuthorData;
+import org.spongepowered.api.service.persistence.InvalidDataException;
+import org.spongepowered.api.text.Texts;
+import org.spongepowered.common.data.SpongeDataProcessor;
+import org.spongepowered.common.data.manipulators.items.SpongeAuthorData;
+
+public class SpongeAuthorProcessor implements SpongeDataProcessor<AuthorData> {
+
+    @Override
+    public Optional<AuthorData> getFrom(DataHolder holder) {
+        if (!(holder instanceof ItemStack)) {
+            return Optional.absent();
+        }
+        if (((ItemStack) holder).getItem() != Items.written_book) {
+            return Optional.absent();
+        }
+        if (!((ItemStack) holder).hasTagCompound()) {
+            return Optional.absent();
+        }
+        if (!((ItemStack) holder).getTagCompound().hasKey("author")) {
+            return Optional.absent();
+        }
+        final String rawAuthor = ((ItemStack) holder).getTagCompound().getString("author");
+        return Optional.of(create().setValue(Texts.of(rawAuthor)));
+    }
+
+    @Override
+    public Optional<AuthorData> fillData(DataHolder holder, AuthorData manipulator, DataPriority priority) {
+        if (holder instanceof org.spongepowered.api.item.inventory.ItemStack) {
+            if (((ItemStack) holder).getItem() != Items.written_book) {
+                return Optional.absent();
+            }
+            switch (checkNotNull(priority)) {
+                case PRE_MERGE:
+                case DATA_MANIPULATOR:
+                    return Optional.of(manipulator);
+                default:
+                    manipulator.setValue(Texts.of(((ItemStack) holder).getTagCompound().getString("author")));
+                    return Optional.of(manipulator);
+            }
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult setData(DataHolder dataHolder, AuthorData manipulator, DataPriority priority) {
+        if (dataHolder instanceof ItemStack && ((ItemStack) dataHolder).getItem() == Items.written_book) {
+            ((ItemStack) dataHolder).setTagInfo("author", new NBTTagString(Texts.toPlain(manipulator.getValue())));
```

Comment:
Needs to switch over the `DataPriority` and populate the `DataTransactionResult` with any pre-existing data that may be replaced.